src/primitive_crypto.cc

```diff
@@ -121,31 +160,53 @@ PRIMITIVE(siphash_get) {
   return result;
 }
 
-AesCbcContext::AesCbcContext(
+AesEcbContext::AesEcbContext(
     SimpleResourceGroup* group,
-    const uint8* key,
-    const uint8* iv,
+    const Blob* key,
     bool encrypt)
   : SimpleResource(group) {
   mbedtls_aes_init(&context_);
   if (encrypt) {
-    mbedtls_aes_setkey_enc(&context_, key, 256);  // 32 byte key.
+    mbedtls_aes_setkey_enc(&context_, key->address(), key->length() * 8); 
   } else {
-    mbedtls_aes_setkey_dec(&context_, key, 256);  // 32 byte key.
+    mbedtls_aes_setkey_dec(&context_, key->address(), key->length() * 8);
   }
-  memcpy(iv_, iv, 16);
 }
 
-AesCbcContext::~AesCbcContext() {
+AesEcbContext::~AesEcbContext() {
   mbedtls_aes_free(&context_);
 }
 
-PRIMITIVE(aes_cbc_init) {
+AesCbcContext::AesCbcContext(
+    SimpleResourceGroup* group,
+    const Blob* key,
+    const uint8* iv,
+    bool encrypt)
+  : AesEcbContext(group, key, encrypt) {
+    memcpy(iv_, iv, 16);
+}
+
+PRIMITIVE(aes_init) {
   ARGS(SimpleResourceGroup, group, Blob, key, Blob, iv, bool, encrypt);
 
-  if (key.length() != 32 || iv.length() != 16) INVALID_ARGUMENT;
+  if ((key.length() != 32 &&
```

Comment:
I'd check the key.length and iv.length separately (using two ifs).

Reply:
The if's have been split into two.